app/src/main/java/net/programmierecke/radiodroid2/players/exoplayer/ExoPlayerWrapper.java

```diff
@@ -98,7 +96,47 @@ public void onReceive(Context context, Intent intent) {
             }
         }
     };
+    final class CustomLoadErrorHandlingPolicy extends DefaultLoadErrorHandlingPolicy {
+        final int MIN_RETRY_DELAY_MS = 10;
+        final SharedPreferences sharedPrefs = PreferenceManager.getDefaultSharedPreferences(context);
 
+        int getSanitizedRetryDelaySettingsMs() {
+            return Math.max(sharedPrefs.getInt("settings_retry_delay", 100), MIN_RETRY_DELAY_MS);
+        }
+
+        @Override
+        public long getRetryDelayMsFor(
+                int dataType,
+                long loadDurationMs,
+                IOException exception,
+                int errorCount) {
+
+            int retryDelay = getSanitizedRetryDelaySettingsMs();
+
+            if (exception instanceof HttpDataSource.InvalidContentTypeException) {
+                stateListener.onPlayerError(R.string.error_play_stream);
+                return C.TIME_UNSET; // Immediately surface error if we cannot play content type
+            }
+
+            if (!Utils.hasAnyConnection(context)) {
+                int resumeWithinS = sharedPrefs.getInt("settings_resume_within", 60);
+                if (resumeWithinS > 0) {
+                    resumeWhenNetworkConnected();
+                    retryDelay = 1000 * resumeWithinS + retryDelay;
+                }
+            }
+
+            if (BuildConfig.DEBUG) {
+                Log.d(TAG, "Providing retry delay of " + retryDelay + "ms for: data type " + dataType + ", load duration: " + loadDurationMs + "ms, error count: " + errorCount + ", exception: " + exception.getClass() + ", message: " + exception.getMessage());
```

Comment:
It's better to break it several lines, the line is too long.

Reply:
Fine with me. I thought it is RadioDroid policy not to break lines :)
We should probably introduce some code styling checks, but I have no experience with that.